mixer/pkg/protobuf/yaml/castUtil.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package yaml
+
+func toFloat(v interface{}) (float64, bool) {
+	switch c := v.(type) {
+	case float32:
+		return float64(c), true
+	case float64:
+		return c, true
+	default:
+		if d, ok := toInt64(v); ok {
```

Comment:
Which scenario does this cover? Is this due to Yaml encoding integer values as integers for floating number fields? (Probably worth a comment, if so).